### DISCUSSION TITLE
Updating to $GITHUB_OUTPUT

### DISCUSSION THREAD

**Discussion by RhyecaRiley-aa** [community/community#36617] [Actions] [answered]
Hello all, 

I am currently trying to convert my teams github actions from using "set-output" to now what is being recommended, $GITHUB_OUTPUT. I cant seem to figure out the appropriate syntax for a specific step in the workflow that looks something like below

        echo "getValue=$( mvn --batch-mode help:evaluate -Dexpression=project.getValue-q -DforceStdout )" >> $GITHUB_OUTPUT
        echo "changeGetValue=$( echo "$getValue" | tr '.' '/' )" >> $GITHUB_OUTPUT
        
To add further clarification our actions looked like this before converting:

        getValue=$( mvn --batch-mode help:evaluate -Dexpression=project.getValue-q -DforceStdout )
        changeGetValue=$( echo $getValue| tr . / )
        echo "INFO: Group ID = ${getValue}"
        echo "::set-output name=getValue::${getValue}"
        echo "::set-output name=changeGetValue::${changeGetValue}"
        
It does not appear to be getting the "getValue" string in the "changeGetValue." I have tried a number of different syntax's including "${getValue}" $getValue '$getValue' and so on. I feel there is something probably very obvious I am missing or just not understanding. Thank you for any advice.

**Comment by airtower-luna** [comment, marked as answer]
You're trying to use a `getValue` environment variable here:

```
    echo "changeGetValue=$( echo "$getValue" | tr '.' '/' )" >> $GITHUB_OUTPUT
```

But such a variable is never created, you only create an output of that name in the previous line:

```
    echo "getValue=$( mvn --batch-mode help:evaluate -Dexpression=project.getValue-q -DforceStdout )" >> $GITHUB_OUTPUT
```

If you want to have the environment variable *and* the output, too (but the former only within the step!), do:

```
    getValue=$( mvn --batch-mode help:evaluate -Dexpression=project.getValue-q -DforceStdout )"
    echo "getValue=${getValue}" >> $GITHUB_OUTPUT
    echo "changeGetValue=$( echo "$getValue" | tr '.' '/' )" >> $GITHUB_OUTPUT
```

Note that this will only work right if each output is only a single line. For multiple lines you need to use the [same format as when setting multi-line environment variables using `GITHUB_ENV`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings).

**Reply by RhyecaRiley-aa** [reply]
Thank you very much! I knew I was overlooking something very obvious (like trying to change an environment variable). I appreciate the help!

**Comment by ossanna16** [comment]
Hey there @RhyecaRiley-aa and thank you for asking a good question! I'm glad you found our community 🙂
